cprover_bindings/src/goto_program/typ.rs

```diff
@@ -71,7 +71,23 @@ pub enum Type {
     /// Packed SIMD vectors
     /// In CBMC/gcc, variables of this type are declared as:
     /// `typ __attribute__((vector_size (size * sizeof(typ)))) var;`
-    Vector { typ: Box<Type>, size: u64 },
+    Vector { data: VectorData },
+}
+
+/// Contains information about SIMD vector types
+#[derive(Debug, Clone)]
+pub struct VectorData {
+    pub typ: Box<Type>,
+    pub size: u64,
+}
+
+impl PartialEq for VectorData {
+    /// Rust allows conversions between SIMD vector types with the same size. By
```

Comment:
I'm a bit curious about this one. Do you have any reference? What kind of conversions are we talking about? I can see conversions between signed and unsigned, but does it allow conversions between float and integer types? Does it allow conversions between types with different sizes?

Reply:
Sorry about the confusion, I shouldn't have used "conversions" here nor in the tests. Conversions as such (i.e., with `as`) don't seem to be allowed for SIMD types.

Unfortunately, there isn't a lot of documentation on this topic. The most complete I've found, at least for this particular SIMD work (the first one, I think) is [this RFC](https://github.com/rust-lang/rfcs/blob/master/text/1199-simd-infrastructure.md), which I've been linking almost everywhere in this PR. But even things like type-checking are loosely defined there, and most of what I've learn has been through trial and error.